akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -111,6 +114,11 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
       /** Network pulls in new frames */
       def onPull(): Unit = updateState(_.onPull())
 
+      override def onDownstreamFinish(): Unit = {
+        frameOutFinished()
+        super.onDownstreamFinish()
```

Comment:
This will trigger cancelStage which now also cancels all substreams with the given cause, and we want our custom `Http2StreamHandling.ConnectionWasAbortedException` as cause for the user stream.